lib/mix/tasks/ecto.gen.migration.ex

```diff
@@ -94,11 +100,30 @@ defmodule Mix.Tasks.Ecto.Gen.Migration do
 
   embed_template :migration, """
   defmodule <%= inspect @mod %> do
-    use Ecto.Migration
+    use <%=  inspect migration_module() %>
 
     def change do
   <%= @change %>
     end
   end
   """
+  
+  @doc """
+  Looks in the application configuration file for a custom migration module to use when generating the migration file.
+  
+  ## Example
+    config :ecto_sql, migration_module: MyApplication.CustomMigrationModule
+
+  If a custom migration module is not defined in the configuration, returns the default Ecto.Migration
+  """
+  defp migration_module() do
+    case Application.get_env(:ecto_sql, :migration_module) do
```

Comment:
Would flipping the case a bit help in readability? Or I think we could handle the case when the module isn't an atom too.

```elixir
case Application.get_env(:ecto_sql, :migration_module) do
  nil -> Ecto.Migration
  migration_module when is_atom(migration_module) -> migration_module
  other -> raise "Expect :migration_module to be an atom, got: #{inspect(other)}"
end
```